bundles/config/org.eclipse.smarthome.config.discovery.usbserial/src/main/java/org/eclipse/smarthome/config/discovery/usbserial/internal/UsbSerialDiscoveryService.java

```diff
@@ -63,12 +65,23 @@
     private final Set<UsbSerialDiscoveryParticipant> discoveryParticipants = new CopyOnWriteArraySet<>();
 
     @Nullable
-    private volatile UsbSerialDiscovery usbSerialDiscovery;
+    private UsbSerialDiscovery usbSerialDiscovery;
 
     public UsbSerialDiscoveryService() {
         super(5);
     }
 
+    @SuppressWarnings("null") // usbSerialDiscovery is not null, as it is set as static reference
```

Comment:
You can mark such injected dependencies as `@NonNullByDefault({})`, see point 12 in https://www.eclipse.org/smarthome/documentation/development/guidelines.html#a-code-style

Afterwards you can remove the `@SuppressWarnings("null")` again.

Reply:
Nice, I wasn't aware of that. Will change obviously :)